### DISCUSSION TITLE
[Features] New Options Vanish-On-join

### DISCUSSION THREAD

**Discussion by MrTokyo12345** [EssentialsX/Essentials#4766] [Ideas and Feature Suggestions]
Pls add the options for automatic vanish on join.

**Comment by kennyrkun** [comment]
Instead it would probably be more in line with the scope of Essentials to simply persist Vanish states as you leave/join the server.

Then, just vanish before you leave.

**Comment by Chris6ix** [comment]
> Pls add the options for automatic vanish on join.

That is already a thing...

Set `allow-silent-join-quit` in your config to `true` and give yourself the permission `essentials.silentjoin.vanish` and you will be vanished when you join
